Fix systemd/init container startup

pavinjosdev opened the pull request:
This PR fixes issue #1060, with Firefox and other complex programs segfaulting due to the following issues:

1. Entering distrobox prior to systemd completing bootup and user integrations.
2.  [pam_systemd](https://www.freedesktop.org/software/systemd/man/latest/pam_systemd.html) not creating new user session when opening default shell or running program as argument to distrobox-enter.

**Test results for hosts**
1. LMDE 6:
- Debian 12 container with init: ✅
- Debian 12 container without init: ✅
2. Fedora 39 Silverblue:
- Debian 12 container with init: ✅
- Debian 12 container without init: ✅
- Fedora 39 container with init: ✅
- Fedora 39 container without init: ✅
- OpenSuse tumbleweed container with init: ❌ 
- Arch Linux container with init: ❌

**Remarks**
1. OpenSuse tumbleweed: init support remains partially broken as `systemd-logind.service` initializes `seat0` but does not initialize a new session when using `su <user>` to login. Using `sudo --user <user> --login` does create a new session and fixes the issue. Cannot use `su` or `sudo` login shell by default as it erases existing environment variables making them not suitable for distrobox purposes.
2. Arch Linux: init support remains permanently broken as `systemd-logind.service` initializes `seat0` but does not initialize a new session when using `su` or `sudo` to login.

Overall, no side effects to existing or new container workflows.

pavinjosdev commented:
Please hold off on this PR.
Arch and Suse creates a new session when using `login` command. Just need to figure out how to execute a command after calling it.

uncomfyhalomacro commented:
i wish this will be merged since my dev container is broken

ZariTen commented:
Yeah me too, my distrobox arch container with Podman is broken since the latest version.

89luca89 commented:
From my testing this also fixes #1078

89luca89 commented:
Thanks a lot @pavinjosdev for the awesome work!

I've adjusted a couple of thing, but seems like now all is working as intended, thanks!

Merging! :rocket:

pavinjosdev commented:
@89luca89 Unfortunately the transfer of code to wait for systemd startup from `distrobox-enter` to `distrobox-init` causes systemd to not assign a session on the first login. This makes sense because we've already used `su` to enter the container prior to systemd startup! 💥

Subsequent logins create a session but I'm getting lots of errors with pipewire and thus losing the audio. I think this is broken in its current implementation.

I just added the waiting part back to `distrobox-enter` and everything works fine.
The implications go beyond just this though. Because as the PAM config was also moved to `distrobox-init`, distros without a pam config for `su` will not be assigned a systemd session on first login.

It looks like [this](https://github.com/89luca89/distrobox/pull/1069/commits/d1a50e6f961b5d752a62e031761dec6a1ddd90f1) needs to be reverted.

89luca89 commented:
@pavinjosdev the latest commits I've pushed waits for login with `systemctl is-system-running` and only after that is done, it prints `container_setup_done`, so that the `enter` can then do its thing

pavinjosdev commented:
> @pavinjosdev the latest commits I've pushed waits for login with `systemctl is-system-running` and only after that is done, it prints `container_setup_done`, so that the `enter` can then do its thing

Ah okay, my bad. I missed distrobox-enter waiting for the `container_setup_done` string from distrobox-init. 🤦
